src/Interactions.jl

```diff
@@ -232,20 +252,36 @@ while `η` controls the direct/reciprocal-space tradeoff in the Ewald summation.
 dipole_dipole(; extent=4, η=0.5) = DipoleDipole(1.0, extent, η)
 
 
-#= Energy and field functions for "simple" interactions that aren't geometry-dependent =#
+#= Energy and field functions for "simple" interactions that aren't geometry-dependent.
+   See Hamiltonian.jl for expectations on `_accum_neggrad!` functions.
+=#
 
-function energy(spins::Array{Vec3}, field::ExternalField)
-    B = field.B
+struct ExternalFieldCPU
+    effBs :: Vector{Vec3}  # |S_b|gᵀB for each basis index b
+end
+
+function ExternalFieldCPU(ext_field::ExternalField, sites_info::Vector{SiteInfo})
+    effBs = [site.g' * site.S * ext_field.B for site in sites_info]
```

Comment:
This is where Bohr magneton should appear?

Also, I guess here we are defining the magnetic moment as `M ~ (S' g)' = g' S`, which explains the transpose. But we should figure out a consistent convention. Note that `g` could possibly (but rarely) have an antisymmetric part, so we need sure to be consistent with the `g` vs `g'` distinction.

Reply:
Yeah, a Bohr magneton needs to appear here and in the construction of `A`.

The transpose here is because we want (ignoring the magneton): 

` E = -∑_i 𝐁 g 𝐒_i`

and I instead precompute an effective `effB = g' * S * 𝐁` so that during the simulations I can just compute:

`E = -∑_i effB ⋅ 𝐬_i`

where `𝐬_i` are the unit vectors for each spin.

Reply:
Looks good. Minor comment, maybe write `E = -∑_i 𝐁^T g 𝐒_i` in the comments, i.e., with the tranpose on `B`. Then it makes sense to define `B_eff^T = 𝐁^T g` or equivalently `B_eff = g^T 𝐁`.